lib/server/openload.py

```diff
@@ -19,76 +19,254 @@
+            oRequest = urllib2.Request(self.__sUrl, sParameters)
+        else:
+            oRequest = urllib2.Request(self.__sUrl)
+
+        for aHeader in self.__aHeaderEntries:
+                for sHeaderKey, sHeaderValue in aHeader.items():
+                    oRequest.add_header(sHeaderKey, sHeaderValue)
+
+        sContent = ''
+
+        try:
+            oResponse = urllib2.urlopen(oRequest, timeout=30)
+            sContent = oResponse.read()
+
+            self.__sResponseHeader = oResponse.info()
+            self.__sRealUrl = oResponse.geturl()
+
+            oResponse.close()
+
+        except urllib2.HTTPError, e:
+            if e.code == 503:
+                if cloudflare.CheckIfActive(e.headers):
+                    cookies = e.headers['Set-Cookie']
+                    cookies = cookies.split(';')[0]
+                    from resources.lib.cloudflare import CloudflareBypass
+                    sContent = CloudflareBypass().GetHtml(self.__sUrl,e.read(),cookies)
+
+                    self.__sResponseHeader = ''
+                    self.__sRealUrl = ''
+
+            if not  sContent:
+                cConfig().error("%s,%s" % (cConfig().getlanguage(30205), self.__sUrl))
+                return ''
+
+        if (self.__bRemoveNewLines == True):
+            sContent = sContent.replace("\n","")
+            sContent = sContent.replace("\r\t","")
+
+        if (self.__bRemoveBreakLines == True):
+            sContent = sContent.replace("&nbsp;","")
+
+        return sContent
+
+    def getHeaderLocationUrl(self):
+        opened = urllib.urlopen(self.__sUrl)
+        return opened.geturl()
+
+class cParser:
+
+    def parseSingleResult(self, sHtmlContent, sPattern):
+        aMatches = re.compile(sPattern).findall(sHtmlContent)
+        if (len(aMatches) == 1):
+                aMatches[0] = self.__replaceSpecialCharacters(aMatches[0])
+                return True, aMatches[0]
+        return False, aMatches
+
+    def __replaceSpecialCharacters(self, sString):
+        return sString.replace('\\/','/').replace('&amp;','&').replace('\xc9','E').replace('&#8211;', '-').replace('&#038;', '&').replace('&rsquo;','\'').replace('\r','').replace('\n','').replace('\t','').replace('&#039;',"'")
+
+    def parse(self, sHtmlContent, sPattern, iMinFoundValue = 1):
+        sHtmlContent = self.__replaceSpecialCharacters(str(sHtmlContent))
+        aMatches = re.compile(sPattern, re.IGNORECASE).findall(sHtmlContent)
+        if (len(aMatches) >= iMinFoundValue):
+            return True, aMatches
+        return False, aMatches
+
+    def replace(self, sPattern, sReplaceString, sValue):
+         return re.sub(sPattern, sReplaceString, sValue)
+
+    def escape(self, sValue):
+        return re.escape(sValue)
+
+    def getNumberFromString(self, sValue):
+        sPattern = "\d+"
+        aMatches = re.findall(sPattern, sValue)
+        if (len(aMatches) > 0):
+            return aMatches[0]
+        return 0
+
+
```

Comment:
These two classes looks like an unnecessary abstraction layer that we don't really need. And if we do so, you should rather implement it in `util` library.